[docs-only]Update testing docs (#5444)
owncloud/ocis: docs/ocis/development/testing.md

Basically we have two sources for feature tests and test suites:

- [oCIS feature test and test suites](https://github.com/owncloud/ocis/tree/master/tests/acceptance/features)
- [tests and test suites transferred from ownCloud, they have prefix coreApi](https://github.com/owncloud/ocis/tree/master/tests/acceptance/features)

At the moment both can be applied to oCIS since the api of oCIS is designed to be compatible with ownCloud.

> As a storage backend we offer oCIS native storage, also called "oCIS". This stores files directly on disk. Along with that we also provide `S3` storage driver.

Reviewer: ```suggestion
As a storage backend, we offer oCIS native storage, also called "ocis". This stores files directly on disk. Along with that we also provide `S3` storage driver.
```

### Run full test suite

The names of the full test suite make targets have the same naming as in the CI pipeline. The available local ocis specific test suites are `apiAccountsHashDifficulty`, `apiArchiver`, `apiContract`, `apiGraph`, `apiSpaces`, `apiSpacesShares`, `apiCors`, `apiAsyncUpload`. They can be run with `oCIS` storage and `S3` storage.

> 

Reviewer 2: Before running test we can add a test log section here instead of down to see our test running while we use the command.

### Run full test suite

> The names of the full test suite make targets have the same naming as in the CI pipeline. The available local ocis specific test suites are `apiAccountsHashDifficulty`, `apiArchiver`, `apiContract`, `apiGraph`, `apiSpaces`, `apiSpacesShares`, `apiCors`, `apiAsyncUpload`. They can be run with `oCIS` storage and `S3` storage.

Reviewer: ```suggestion
The names of the full test suite make targets have the same naming as in the CI pipeline. The available local ocis specific test suites are `apiAccountsHashDifficulty`, `apiArchiver`, `apiContract`, `apiGraph`, `apiSpaces`, `apiSpacesShares`, `apiCors`, `apiAsyncUpload`. They can be run with `ocis` storage and `S3` storage.
```

> Note: In order to see the tests log attach `show-test-logs` in the command

For example `make -C tests/acceptance/docker localApiTests-apiAccountsHashDifficulty-ocis` runs the same tests as the `localApiTests-apiAccountsHashDifficulty-ocis` CI pipeline, which runs the oCIS test suite "apiAccountsHashDifficulty" against an oCIS with oCIS storage.

For example `make -C tests/acceptance/docker localApiTests-apiAccountsHashDifficulty-s3ng` runs the oCIS test suite "apiAccountsHashDifficulty" against an oCIS with s3 storage.

> Note: To run the tests from `apiAsyncUpload` suite you need to provide extra environment variable `POSTPROCESSING_DELAY`

For example `make -C tests/acceptance/docker Core-API-Tests-ocis-storage-3` runs the same tests as the `Core-API-Tests-ocis-storage-3` CI pipeline, which runs the third (out of ten) test suite transferred from ownCloud against an oCIS with ocis storage.

> For example `make -C tests/acceptance/docker Core-API-Tests-ocis-s3ng-3` runs the third (out of ten) test suite transferred from ownCloud against an oCIS with s3 storage.

Reviewer: ```suggestion
For example `make -C tests/acceptance/docker Core-API-Tests-s3ng-storage-3` runs the third (out of ten) test suite transferred from ownCloud against an oCIS with s3 storage.
```

> For example `make -C tests/acceptance/docker localApiTests-apiAccountsHashDifficulty-ocis` runs the same tests as the `localApiTests-apiAccountsHashDifficulty-ocis` CI pipeline, which runs the oCIS test suite "apiAccountsHashDifficulty" against an oCIS with oCIS storage.

Reviewer 3: Maybe specify which sets they are?
```
"apiAccountsHashDifficulty",
 "apiArchiver",
"apiContract",
 "apiGraph",
"apiSpaces",
"apiSpacesShares"
```

Reviewer 3: and with which of the possible storage we can run tests

> make -C tests/acceptance/docker test-ocis-feature-ocis-storage BEHAT_FEATURE='tests/acceptance/features/apiAccountsHashDifficulty/addUser.feature:20'

Reviewer 2: ```suggestion
make -C tests/acceptance/docker test-ocis-feature-ocis-storage BEHAT_FEATURE='tests/acceptance/features/apiAccountsHashDifficulty/addUser.feature:21'
```

```

Similarly, with S3 storage,
- run a whole feature
```
make -C tests/acceptance/docker test-ocis-feature-s3ng-storage BEHAT_FEATURE='tests/acceptance/features/apiAccountsHashDifficulty/addUser.feature'
```

- run a single scenario
```

> make -C tests/acceptance/docker test-ocis-feature-s3ng-storage BEHAT_FEATURE='tests/acceptance/features/apiAccountsHashDifficulty/addUser.feature:20'

Reviewer 2: ```suggestion
make -C tests/acceptance/docker test-ocis-feature-s3ng-storage BEHAT_FEATURE='tests/acceptance/features/apiAccountsHashDifficulty/addUser.feature:21'
```

```

In the same way for the tests transferred from oc10 can be run as
- run a whole feature
```
make -C tests/acceptance/docker test-oc10-feature-ocis-storage BEHAT_FEATURE='tests/acceptance/features/coreApiAuth/webDavAuth.feature'
```

- run a single scenario
```
make -C tests/acceptance/docker test-oc10-feature-ocis-storage BEHAT_FEATURE='tests/acceptance/features/coreApiAuth/webDavAuth.feature:13'
```

> > Note: the tests transferred fron oc10 starts with coreApi

Author: ```suggestion
> Note: the tests transferred from oc10 starts with coreApi
```

Reviewer 4: ```suggestion
> Note: the tests transferred from oc10 start with coreApi
```
